packages/hub/node-tests/routes/profiles-test.ts

```diff
@@ -0,0 +1,669 @@
+          detail: 'This ID is already taken. Please choose another one',
+        })
+        .expect('Content-Type', 'application/vnd.api+json');
+
+      const slug2 = 'mandello2';
+      await request()
+        .get(`/api/profiles/validate-slug/${slug2}`)
+        .set('Authorization', 'Bearer abc123--def456--ghi789')
+        .set('Content-Type', 'application/vnd.api+json')
+        .expect(200)
+        .expect({
+          slugAvailable: true,
+          detail: 'ID is available',
+        })
+        .expect('Content-Type', 'application/vnd.api+json');
+    });
+  });
+
+  describe('GET /api/profiles/short-id/:id', function () {
+    const fullUuid = 'a7eeb098-f8d6-4926-a47b-c320b7375d6b';
+    // shortened using short-uuid, originally 'a7eeb098-f8d6-4926-a47b-c320b7375d6b'
+    // this endpoint should convert it back
+    const shortenedUuid = 'mJKhNVKyAUgScu1dysmR8R';
+    let { request, getPrisma } = setupHub(this);
+
+    this.beforeEach(async function () {
+      let prisma = await getPrisma();
+      await prisma.profile.create({
+        data: {
+          id: fullUuid,
+          name: 'Merchie!',
+          slug: 'slug',
+          color: 'red',
+          textColor: 'purple',
+          ownerAddress: '0x2f58630CA445Ab1a6DE2Bb9892AA2e1d60876C13',
+        },
+      });
+    });
+
+    it('returns 404 if short id cannot be resolved', async function () {
+      await request()
+        .get(`/api/profiles/short-id/now6HxUjmBfDwTbF76GVHz`)
+        .set('Content-Type', 'application/vnd.api+json')
+        .expect(404);
+    });
+
+    it('can return a profile customization', async function () {
+      await request()
+        .get(`/api/profiles/short-id/${shortenedUuid}`)
+        .set('Content-Type', 'application/vnd.api+json')
+        .expect({
+          meta: { network: 'sokol' },
+          data: {
+            id: fullUuid,
+            type: 'profiles',
+            attributes: {
+              did: 'did:cardstack:1mmJKhNVKyAUgScu1dysmR8R728e839d5d105bff',
+              name: 'Merchie!',
+              slug: 'slug',
+              color: 'red',
+              'text-color': 'purple',
+              'owner-address': '0x2f58630CA445Ab1a6DE2Bb9892AA2e1d60876C13',
+              links: [],
+              'profile-description': null,
+              'profile-image-url': null,
+            },
+          },
+        })
+        .expect('Content-Type', 'application/vnd.api+json');
+    });
+  });
+  describe('POST /api/profiles', function () {
+    this.beforeEach(function () {
+      registry(this).register('authentication-utils', StubAuthenticationUtils);
+    });
+
+    let { request } = setupHub(this);
+
+    it('persists profile', async function () {
+      const payload = {
+        data: {
+          type: 'profiles',
+          attributes: {
+            name: 'Satoshi Nakamoto',
+            slug: 'satoshi',
+            color: 'ff0000',
+            'text-color': 'ffffff',
+            'owner-address': '0x00000000000',
```

Comment:
Pre-existing problem - `ownerAddress` is derived from the auth token, so we should probably remove it from payloads used for profile creation tests?

Reply:
We could also consider changing the way that we handle addresses because it's possible to end up with a "stale" auth token/ an auth token for another address - so the auth token's address is checked against the supplied `owner-address` and we return a 401 if they are not the same.